### PR TITLE
Optionally run perltidy on save

### PR DESCRIPTION
**Pull request by mndrix**
This creates a new setting ("Tidy On Save").  When that setting is true,
perltidy is automatically run for each Perl file immediately after it's
saved.

### CODE REVIEW

**Comment by mndrix** [comment]
I've rebased this pull request on top of the most recent master branch.

@kraih does this have a chance of being merged?

**Comment by kraih** [comment]
I'd like to have tests first, before adding features i don't use myself.

File: index.js [hunk 1]
```diff
@@ -44,3 +49,17 @@ function perlTidy(path, before, cb) {
     after += chunk;
   });
 }
+
+function onSave() {
+  var tidyOnSave = atom.config.get('perltidy.tidyOnSave');
+  if (!tidyOnSave) return;
+
+  var editor = atom.workspaceView.getActiveView().editor;
```
Comment (kraih) [inline index.js]: Pretty sure some of the APIs used are deprecated.

**Comment by mndrix** [comment]
I've fixed the deprecated APIs.  I also now handle the package being disabled/re-enabled by disposing of subscriptions when appropriate.

I understand your desire for tests.  At this point, I'm unlikely to devote the [necessary effort](https://github.com/joefitzgerald/go-plus/blob/42e8555fd3f606e3ea89c8689c618320153f2266/spec/gofmt-spec.coffee). I use this feature daily and that's good enough for me.  No hard feelings if you decide to close the pull request.

**Comment by kraih** [comment]
Thanks for your effort, but it doesn't look like we'll get tests anytime soon, so i'll have to reject this feature for now.

**Pull request closed without merging**